### PR TITLE
Update TextLayer example

### PR DESCRIPTION
**Pull request by Pessimistress**
For #7635 

#### Change List
- Rename the tagmap example to text
- Drop 3rd party library, use CollideExtension to prevent overlap
- New dataset

### CODE REVIEW

**Review by chrisgervang — APPROVED** [review]
Nice way to showcase the new feature! Looks great.

File: examples/website/text/app.jsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    id: 'world-cities',
+    data,
+    characterSet: 'auto',
+    fontSettings: {
+      buffer: 8
+    },
+
+    // TextLayer options
+    getText: d => d.name,
+    getPosition: d => [d.longitude, d.latitude],
```
Comment (felixpalmer) [inline examples/website/text/app.jsx:63]: `csv()` returns these as strings, I had to change to floats to get it running locally

Comment (felixpalmer) [inline examples/website/text/app.jsx:63]:
```suggestion
    getPosition: d => [parseFloat(d.longitude), parseFloat(d.latitude)],
```

File: examples/website/text/app.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      <Map reuseMaps mapLib={maplibregl} mapStyle={mapStyle} preventStyleDiffing={true} />
+    </DeckGL>
+  );
+}
+
+export function renderToDOM(container) {
+  const root = createRoot(container);
+  root.render(<App />);
+
+  csv(DATA_URL, (error, data) => {
```
Comment (felixpalmer) [inline examples/website/text/app.jsx:99]: Any reason we can't use CSVLoader here?

File: examples/website/text/app.jsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import DeckGL from '@deck.gl/react';
+import {MapView} from '@deck.gl/core';
+import {TextLayer} from '@deck.gl/layers';
+import {CollideExtension} from '@deck.gl/extensions';
+import {scaleLinear} from 'd3-scale';
+
+import {csv} from 'd3-request';
+
+// Sample data
+const DATA_URL = 'https://raw.githubusercontent.com/visgl/deck.gl-data/master/examples/text-layer/cities-1000.csv';
```
Comment (felixpalmer) [inline examples/website/text/app.jsx:16]: Performance feels worse than https://pessimistress.github.io/experiments/collide/, is it the same dataset?

**Review by felixpalmer — CHANGES_REQUESTED** [review]

File: examples/website/text/app.jsx (hunk 3 above)
Comment (Pessimistress) [reply examples/website/text/app.jsx:16]: It's the same. React in development mode usually has worse perf.

**Review by felixpalmer — APPROVED** [review]

**Pull request merged**